uiowahjmjohnsonselt2024/projectdirectory-selt_2024_team_009
# IMPL: added wallet functionalty,new member bonus,buying shards

## What does this PR do?
This PR introduces the following functionality:
- Created wallet functionality for managing in-game currency (shards).
- Enabled buying items using shards purchased.
- New signups now receive a bonus of 500 shards.
- Added the ability to select inventory when starting the game.
- Implemented "Open Gem" feature for additional rewards.
- Added necessary initializers to set up the game.

## Related Issues


## Testing
Steps to test the features:
1. Create a new account and verify the 500 shards bonus is applied.
2. Purchase shards and use them to buy in-game items.
3. Select inventory during the game setup process.
4. Test the "Open Gem" feature for functionality and expected behavior.
5. Confirm all initializers are working as intended to set up the game.

## Checklist

## Extra Information
- Includes initial setup for wallet and shard-related transactions.
- Performance checks and additional screenshots (if applicable) can be added here.
